nucliadb_texts2/src/writer.rs

```diff
@@ -296,6 +296,23 @@ impl TextWriterService {
             self.schema.status => resource.status as u64,
         );
 
+        let public_to_all_groups = resource.security.is_none()
+            || resource.security.as_ref().unwrap().access_groups.is_empty();
```

Comment:
Although this logic is safe I am not a huge fan of the behaviour it hides, basically it assumes lazy evaluation and that is why unwrapping in the right-hand side is safe. I would suggest the following:
```rust
        let public_to_all_groups = resource
            .security
            .as_ref()
            .map_or(true, |i| i.access_groups.is_empty());
```